STORY18: Include profile pages and posts in app

s1gr4y opened the pull request.

KWagnerCS committed:
Updated Radius Filter component

s1gr4y committed:
STORY19: added enpoint and ability to get a specific post by id

s1gr4y committed:
STORY18: Added routing for geting all posts from specific profile

KWagnerCS reviewed (approved):
Changes:
- GetUserPosts
- GetPostById
- Routes:
 - /profile/{id}
 - /posts/{id}

KWagnerCS committed:
Update .gitignore

KWagnerCS committed:
stashed changes released

KWagnerCS committed:
Merge branch 'scratch/STORY19' of https://github.com/bryetz/SpotLight into scratch/STORY19

KWagnerCS committed:
Added frontend functionality for clickable posts and sharing posts

- fixed backend query for getPostByID
- added deleting functionality
- added sharing using share web api
- added dynamic post/[id] page
- updated api.ts
- added more props to Feed component

KWagnerCS committed:
Merge branch 'scratch/STORY19' into scratch/STORY18

KWagnerCS committed:
Implemented frontend for profile viewing

- new profile/[id] dynamic page
- clickable usernames in posts
- clickable comment usernames
- can use header "view profile" button
- fixed backend issues
- added user_id to comments in backend

s1gr4y committed:
STORY18: Added tests for getting user posts and specific posts

s1gr4y committed:
Merge branch 'main' into scratch/STORY18
